### PR TITLE
Add MAnet 

### PR DESCRIPTION
**Pull request by azkalot1**
Paper: https://ieeexplore.ieee.org/abstract/document/9201310
Seems to work better than Unet, similar to Unet++ but faster

An implementation is similar to the original in the paper, except for encoder part.

### CODE REVIEW

**Comment by qubvel** [comment]
A few more things:
 - Update README.md
 - Update docs/models.rst
 - Check model docstring to be constistent with existing models

May be write a little bit more in description of the model in docstring

File: README.md [hunk 1]
```diff
@@ -76,6 +76,7 @@ Congratulations! You are done! Now you can train your model with your favorite f
 #### Architectures <a name="architectires"></a>
  - Unet [[paper](https://arxiv.org/abs/1505.04597)] [[docs](https://smp.readthedocs.io/en/latest/models.html#unet)]
  - Unet++ [[paper](https://arxiv.org/pdf/1807.10165.pdf)] [[docs](https://smp.readthedocs.io/en/latest/models.html#id2)]
+ - MAnet [[paper](https://ieeexplore.ieee.org/abstract/document/9201310)] [[docs](https://smp.readthedocs.io/en/latest/models.html#id2)]
```
Comment (qubvel) [inline README.md:79]: 1. Fix link: https://smp.readthedocs.io/en/latest/models.html#manet
2. Change number of archs on top of the README

File: segmentation_models_pytorch/__init__.py [hunk 2]
```diff
@@ -24,9 +25,9 @@ def create_model(
     **kwargs,
 ) -> torch.nn.Module:
     """Models wrapper. Allows to create any model just with parametes
-    
+
     """
-    
+
     archs = [Unet, UnetPlusPlus, Linknet, FPN, PSPNet, DeepLabV3, DeepLabV3Plus, PAN]
```
Comment (qubvel) [inline segmentation_models_pytorch/__init__.py:31]: Add here MANet

File: segmentation_models_pytorch/manet/model.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+
+class MAnet(SegmentationModel):
+    """MAnet_ :  Multi-scale Attention Net.
+       The MA-Net can capture rich contextual dependencies based on the attention mechanism, using two blocks:
+       Position-wise Attention Block (PAB, which captures the spatial dependencies between pixels in a global view)
+       and Multi-scale Fusion Attention Block (MFAB, which  captures the channel dependencies between any feature map by
+       multi-scale semantic feature fusion)
+
+    Args:
```
Comment (qubvel) [inline segmentation_models_pytorch/manet/model.py:15]: Args docs from Unet (just copy-paste if all params are the same):

```
 Args:
        encoder_name: Name of the classification model that will be used as an encoder (a.k.a backbone)
            to extract features of different spatial resolution
        encoder_depth: A number of stages used in encoder in range [3, 5]. Each stage generate features 
            two times smaller in spatial dimentions than previous one (e.g. for depth 0 we will have features 
            with shapes [(N, C, H, W),], for depth 1 - [(N, C, H, W), (N, C, H // 2, W // 2)] and so on).
            Default is 5
        encoder_weights: One of **None** (random initialization), **"imagenet"** (pre-training on ImageNet) and 
            other pretrained weights (see table with available weights for each encoder_name)
        decoder_channels: List of integers which specify **in_channels** parameter for convolutions used in decoder.
            Lenght of the list should be the same as **encoder_depth**
        decoder_use_batchnorm: If **True**, BatchNorm2d layer between Conv2D and Activation layers
            is used. If **"inplace"** InplaceABN will be used, allows to decrease memory consumption.
            Avaliable options are **True, False, "inplace"**
        decoder_attention_type: Attention module used in decoder of the model. Avaliable options are **None** and **scse**.
            SCSE paper - https://arxiv.org/abs/1808.08127
        in_channels: A number of input channels for the model, default is 3 (RGB images)
        classes: A number of classes for output mask (or you can think as a number of channels of output mask)
        activation: An activation function to apply after the final convolution layer.
            Avaliable options are **"sigmoid"**, **"softmax"**, **"logsoftmax"**, **"identity"**, **callable** and **None**.
            Default is **None**
        aux_params: Dictionary with parameters of the auxiliary output (classification head). Auxiliary output is build 
            on top of encoder if **aux_params** is not **None** (default). Supported params:
                - classes (int): A number of classes
                - pooling (str): One of "max", "avg". Default is "avg"
                - dropout (float): Dropout factor in [0, 1)
                - activation (str): An activation function to apply "sigmoid"/"softmax" (could be **None** to return logits)
```

File: segmentation_models_pytorch/manet/model.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    """
+
+    def __init__(
+        self,
+        encoder_name: str = "resnet34",
+        encoder_depth: int = 5,
+        encoder_weights: str = "imagenet",
+        decoder_use_batchnorm: bool = True,
+        decoder_channels: List[int] = (256, 128, 64, 32, 16),
+        decoder_im_channels: int = 64,
```
Comment (qubvel) [inline segmentation_models_pytorch/manet/model.py:54]: Maybe rename to `decoder_pab_channels` if it is a number of channels in PAB block?

**Comment by qubvel** [comment]
:+1:

**Comment by ynhuhu** [comment]
Thanks sharing！
I installed segmentation_models_pytorch 
model = smp.MAnet(encoder_name=ENCODER, encoder_weights=ENCODER_WEIGHTS,
                        classes=len(CLASSES), 
                            activation=ACTIVATION,
                        )

module 'segmentation_models_pytorch' has no attribute 'MAnet'
Why？

**Comment by qubvel** [comment]
Install the latest version, look at the instructions in README.md

**Comment by pritesh-aidash** [comment]
I have the latest version. MAnet still does not show up.

**Comment by fmellomascarenhas** [comment]
@pritesh-aidash @qubvel same thing for me.

So I did 
```
import segmentation_models_pytorch as smp
smp??
```

And it showed
```
from .unet import Unet
from .unetplusplus import UnetPlusPlus
from .manet import MAnet
from .linknet import Linknet
from .fpn import FPN
[...]
```

trying to do smp.MAnet doesnt work, but smp.manet.MAnet does

**Pull request merged**